src/components/ec-dropdown-search/ec-dropdown-search.vue

```diff
@@ -343,6 +347,19 @@ const filteredItems = computed(() => {
   }
 
   return props.items.filter((item) => {
+    if (props.searchFields?.length) {
+      let hasSearchValue = false;
+      for (const searchField of props.searchFields) {
+        if (item[searchField]) {
+          const searchValue = removeDiacritics(item[searchField].trim().toLowerCase());
+          if (searchValue.includes(sanitisedText)) {
+            hasSearchValue = true;
+            break;
+          }
+        }
+      }
+      if (hasSearchValue) return hasSearchValue;
+    }
```

Comment:
The performance of this is getting worse and worse. We need to removeDiacritics from every field every time the user presses a key. We need some kind of cache here and calculate the searchable value only once per item.

We could have a separate computed prop which updates only when `items` changed, something like this:

```
const indexedItems = computed(() => {
	return new WeakMap(items.value.map(item => {
    return [item, makeIndex(item, props.searchFields ?? ['text'])]; // defaults to text field of the item if not passed
  }));
});
  
function makeIndex(item, searchFields) {
  return searchFields.map(searchField => {
    return removeDiacritics(item[searchField] ?? '').toLowerCase();
  }).join('\u00A0'); // all values have their values sanitised and concatenated into a single string separated by a value that user can't type on the keyboard.
}
```
Then we can use the cache to retrieve sanitised text of each item and search there:
```
const filteredItems = computed(() => {
  const sanitisedText = removeDiacritics(filterText.value.toLowerCase());
  if (!sanitisedText) {
    return props.items;
  }

  return props.items.filter((item) => {
    const indexedText = indexedItems.value.get(i);
    return indexedText.includes(sanitisedText);
  });
});
```

And this covers both scenarios when searchFields are set or not.